homeassistant/components/fritz/common.py

```diff
@@ -663,10 +663,12 @@ def get_wlan_configuration(self, index: int) -> dict[str, Any]:
 
         return self._service_call_action("WLANConfiguration", str(index), "GetInfo")
 
-    def get_wan_dsl_interface_config(self) -> dict[str, Any]:
-        """Call WANDSLInterfaceConfig service."""
+    def get_wan_link_properties(self) -> dict[str, Any]:
+        """Call WANCommonInterfaceConfig service."""
 
-        return self._service_call_action("WANDSLInterfaceConfig", "1", "GetInfo")
+        return self._service_call_action(
+            "WANCommonInterfaceConfig", "1", "GetCommonLinkProperties"
+        )
```

Comment:
We need to investigate what happens when there is no WAN ( like for example when [Setting up the FRITZ!Box as an IP client](https://en.avm.de/service/knowledge-base/dok/FRITZ-Box-7590/3244_Setting-up-the-FRITZ-Box-as-an-IP-client/)

Seems "None" is not a allowed reply: <https://avm.de/fileadmin/user_upload/Global/Service/Schnittstellen/wancommonifconfigSCPD.pdf>